java-checks/src/main/java/org/sonar/java/checks/StaticMultithreadedUnsafeFieldsCheck.java

```diff
@@ -61,8 +68,20 @@ public void visitNode(Tree tree) {
   }
 
   private static boolean isForbiddenType(VariableTree variableTree) {
+    if (isForbiddenType(variableTree.type().symbolType())) {
+      return true;
+    }
     ExpressionTree initializer = variableTree.initializer();
-    return isForbiddenType(variableTree.type().symbolType()) || (initializer != null && !initializer.is(Tree.Kind.NULL_LITERAL) && isForbiddenType(initializer.symbolType()));
+    if (initializer == null) {
+      return false;
+    }
+    if (initializer.is(Tree.Kind.NULL_LITERAL)) {
+      return false;
+    }
+    if (initializer.is(Tree.Kind.METHOD_INVOCATION)) {
+      return GET_DATE_INSTANCE.matches((MethodInvocationTree) initializer);
```

Comment:
This will break the case where the initializer is a method call returning a forbidden type (which should be added as a test case since currently there are only test where a constructor is called rather than a method). The check for `isForbiddenType(initializer.symbolType())` should come before this one. Or alternatively this one should only return if the method matcher matches, i.e.:

    if (initializer.is(Tree.Kind.METHOD_INVOCATION) && GET_DATE_INSTANCE.matches((MethodInvocationTree) initializer)) {
      return true;
    }